js/wp-seo-wordpress-scraper.js

```diff
@@ -0,0 +1,394 @@
+    jQuery.post( ajaxurl, {
+            action: 'wpseo_replace_vars',
+            string: srcObj.replaceableVar,
+            post_id: jQuery( '#post_ID' ).val(),
+            _wpnonce: wpseoMetaboxL10n.wpseo_replace_vars_nonce
+        }, function( data ) {
+            if ( data ) {
+				YoastSEO_loader.source.replacedVars[srcObj.replaceableVar] = data;
+				YoastSEO_loader.source.replaceVariables (srcObj.textString, srcObj.type, srcObj.object);
+            }
+        }
+    );
+};
+
+/**
+ * calls the eventbinders.
+ */
+YoastSEO_WordPressScraper.prototype.bindElementEvents = function() {
+    this.snippetPreviewEventBinder();
+    this.inputElementEventBinder();
+	document.getElementById("yoast_wpseo_focuskw").addEventListener("keydown", this.refObj.snippetPreview.disableEnter);
+};
+
+/**
+ * binds the getinputfieldsdata to the snippetelements.
+ */
+YoastSEO_WordPressScraper.prototype.snippetPreviewEventBinder = function() {
+    var elems = ["cite", "meta", "title"];
+    for (var i = 0; i < elems.length; i++){
+        document.getElementById("snippet_"+elems[i]).addEventListener("focus", this.getInputFieldsData);
+		document.getElementById("snippet_"+elems[i]).addEventListener("keydown", this.refObj.snippetPreview.disableEnter);
+		document.getElementById("snippet_"+elems[i]).addEventListener("blur", this.refObj.snippetPreview.checkTextLength);
+		//textFeedback is given on input (when user types or pastests), but also on focus. If a string that is too long is being recalled
+		//from the saved values, it gets the correct classname right away.
+		document.getElementById("snippet_"+elems[i]).addEventListener("input", this.refObj.snippetPreview.textFeedback);
+		document.getElementById("snippet_"+elems[i]).addEventListener("focus", this.refObj.snippetPreview.textFeedback);
+		//shows edit icon by hovering over element
+		document.getElementById("snippet_"+elems[i]).addEventListener("mouseover", this.refObj.snippetPreview.showEditIcon);
+		//hides the edit icon onmouseout, on focus and on keyup. If user clicks or types AND moves his mouse, the edit icon could return while editting
+		//by binding to these 3 events
+		document.getElementById("snippet_"+elems[i]).addEventListener("mouseout", this.refObj.snippetPreview.hideEditIcon);
+		document.getElementById("snippet_"+elems[i]).addEventListener("focus", this.refObj.snippetPreview.hideEditIcon);
+		document.getElementById("snippet_"+elems[i]).addEventListener("keyup", this.refObj.snippetPreview.hideEditIcon);
+    }
+};
+
+/**
+ * bins the renewData function on the change of inputelements.
+ */
+YoastSEO_WordPressScraper.prototype.inputElementEventBinder = function() {
+    var elems = ["excerpt", "content", "editable-post-name", "yoast_wpseo_focuskw"];
+    for (var i = 0; i < elems.length; i++){
+		var elem = document.getElementById(elems[i]);
+		if(elem !== null) {
+			document.getElementById(elems[i]).addEventListener("change", this.renewData);
```

Comment:
You cache the element in the `elem` variable here but then call `getElementById` again.